app/routes.py

```diff
@@ -0,0 +1,291 @@
+        return ({"message": f"Customer {customer_id} was not found"}),404
+
+    if request.method == "GET":
+        rentals_response = []
+        for video in customer.videos:
+            rentals_response.append({
+                "title": video.title,
+                "release_date": video.release_date,
+                "due_date":Rental.query.get(customer_id).due_date
+            })
+        return jsonify(rentals_response)
+
+
+@rental_bp.route("/check-in", methods = ["POST"])
+def handle_checkin():
+    if request.method == "POST":
+        request_body = request.get_json()
+        if "customer_id" not in request_body:
+            return ({
+                "details": "Request body must include customer_id."
+            }),400
+        elif "video_id" not in request_body:
+            return ({
+                "details": "Request body must include video_id."
+            }),400
+        #check that the video exists before attempting to return the rental
+        target_video = Video.query.get(request_body['video_id'])
+        if not target_video:
+            return "Video_id doesn't exist", 404
+        #check that the customer exists before attempting to return the rental
+        target_customer = Customer.query.get(request_body['customer_id'])
+        if not target_customer:
+            return "Customer_id doesn't exist", 404
+        
+        return_rental = Rental.query.filter_by(
+            customer_id = request_body["customer_id"],
+            video_id = request_body["video_id"]
+        ).first()
+        if return_rental is None:
+            return {"message": f"No outstanding rentals for customer {request_body['customer_id']} and video {request_body['video_id']}"}, 400
+        else:
+    
+            db.session.delete(return_rental)
+     
+        total_inventory = Video.query.get(return_rental.video_id).total_inventory
+        existing_rentals = rentals_associated(return_rental)
+        available_inventory = total_inventory - existing_rentals
+        
+        return ({
+            "customer_id": return_rental.customer_id,
+            "video_id": return_rental.video_id,
+            "videos_checked_out_count": videos_checked_out_count(return_rental),
+            "available_inventory": available_inventory
+        }),200
+
+      
+@videos_bp.route("/<video_id>/rentals", methods = ["GET"])
+def handle_rentals_by_id(video_id):
+    if not video_id.isnumeric():
+        return { "Error": f"{video_id} must be numeric."}, 400
+    video_id = int(video_id)
+    video = Video.query.get(video_id)
+    if not video:
+        return ({"message":f"Video {video_id} was not found"}), 404
+    rentals_response = []
+    for customer in video.customers:
+        rentals_response.append({
+            "name": customer.name
+        })
+
+    return jsonify(rentals_response)
+
+#HELPER FUNCTIONS
+
+def rentals_associated(rental):
+    return len(db.session.query(Rental).filter(Rental.video_id == rental.video_id).all())
```

Comment:
This might be better done with a relationship, so you can do `video.rentals`. Or `rental.video.rentals` to get all the other rentals.